Buddies/Utilities/LocationPersistence.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  LocationPersistence.swift
+//  Buddies
+//
+//  Created by Jake Thurman on 3/13/19.
+//  Copyright © 2019 Jack and the Beans. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import FirebaseFirestore
+
+class LocationPersistence : NSObject, CLLocationManagerDelegate {
+    static let instance = LocationPersistence()
+
+    let manager = CLLocationManager()
+    var cancelUserListener: Canceler!
+    var user: LoggedInUser?
+    
+    deinit {
+        cancelUserListener()
+    }
+    
+    init(dataAccessor: DataAccessor = DataAccessor.instance) {
+        super.init()
+        
+        manager.desiredAccuracy = kCLLocationAccuracyBest
+        manager.delegate = self
+        
+        cancelUserListener = dataAccessor.useLoggedInUser { user in
+            self.user = user
+            
+            // On user change, record where we are imediately
```

Comment:
This seems very backwards - shouldn't it (and the code) be "on location change, record where we are immediately"?

Reply:
It's also watched below when moving

Reply:
this for onlogin